api/api.py

```diff
@@ -0,0 +1,659 @@
+
+    tag_name: str
+
+    class Config:
+        orm_mode = True
+
+
+class VerdictModel(BaseModel):
+    """Pydantic model for the verdict table"""
+
+    verdict: str
+
+    class Config:
+        orm_mode = True
+
+
+class LawyerModel(BaseModel):
+    """Pydantic model for the lawyer table"""
+
+    lawyer_name: Optional[str]
+    law_firm: Optional[LawFirmModel]
+
+    class Config:
+        orm_mode = True
+
+
+class ParticipantAssignmentModel(BaseModel):
+    """Pydantic model for the participant_assignment table"""
+
+    is_defendant: Optional[bool]
+    participant_name: Optional[str]
+    lawyer_name: Optional[str]
+    law_firm_name: Optional[str]
+
+    class Config:
+        orm_mode = True
+
+
+class ParticipantAssignmentWithCourtCaseModel(BaseModel):
+    """Pydantic model for the participant_assignment table with court_case_id"""
+
+    court_case_id: str
+    is_defendant: Optional[bool]
+    participant_name: Optional[str]
+    lawyer_name: Optional[str]
+    law_firm_name: Optional[str]
+
+    class Config:
+        orm_mode = True
+
+
+class CourtCaseModel(BaseModel):
+    """Pydantic model for the court_case table"""
+
+    court_case_id: str
+    summary: Optional[str]
+    title: Optional[str]
+    court_date: Optional[date]
+    case_number: Optional[str]
+    case_url: Optional[str]
+    verdict_summary: Optional[str]
+    court: Optional[CourtModel]
+    verdict: Optional[VerdictModel]
+    tags: List[TagModel]
+    judges: List[JudgeModel]
+    participant_assignments: List[ParticipantAssignmentModel]
+
+    class Config:
+        orm_mode = True
+
+
+def no_matches(response: Response, endpoint: str) -> JSONResponse:
+    """Returns a JSON response with a message if no matches are found"""
+    response.status_code = status.HTTP_200_OK
+    return JSONResponse(
+        {"message": f"No matching {endpoint} found matching query parameters."}
+    )
+
+
+def validate_query_params(params: dict, query_param_list: list) -> None:
+    unsupported_params = [
+        query_param
+        for query_param in params.keys()
+        if query_param not in query_param_list
+    ]
+
+    if unsupported_params:
+        raise HTTPException(
+            status_code=400,
+            detail=f"Query parameter(s) '{', '.join(unsupported_params)}' are not supported. Supported query parameters are {query_param_list}",
+        )
+
+
+@app.get("/")
+def get_api_overview() -> JSONResponse:
+    return JSONResponse(
+        {
+            "message": "Welcome to the Justice Lens API",
+            "endpoints": {
+                "/courts/": "read_courts",
+                "/judges/": "read_judges",
+                "/lawyers/": "read_lawyers",
+                "/law_firms/": "read_law_firms",
+                "/participants/": "read_participants",
+                "/tags/": "read_tags",
+                "/verdicts/": "read_verdicts",
+                "/court_cases/": "read_court_cases",
+            },
```

Comment:
I would maybe add a parameters key that explains things like how the limit parameter works, e.g. defaults to 100, set to -1 for all. Although this is in the docs, it would be nice for the potential engineers accessing this to deal with as little documentation as possible.

Reply:
Great point, I have added this mention in the root endpoint messages.